### ISSUE TITLE
Adding color to a cell messes up spacing

### ISSUE THREAD

**Issue by abierbaum** [ayonious/console-table-printer#314]
See https://github.com/ayonious/console-table-printer/issues/57 for another report of similar bug.

If I do something like:

```
const p = new Table();
p.addRows(
  [
    { index: 3, text: chalk.red("text"), value: 100 },
    { index: 4, text: 'short', value: 300 },
  ],
);
p.printTable();
```

Then the spacing for the cell is off because it's width is incorrectly calculated.

**Comment by ayonious** [comment]
@abierbaum there is no feature currently to do cell coloring. 
You can color row/column but not cell.

**Comment by abierbaum** [comment]
@ayonious Understood.  I guess a better way to state this is there is a bug where if you put text into a cell that has ANSI characters, the width of the text is computed incorrectly.

I think it could be corrected by changing code that uses `.length` like here: https://github.com/ayonious/console-table-printer/blob/0ba9b865e53e65daabbd2e1cad0d33508398433c/src/utils/table-helpers.ts#L39

To use something like strip-ansi [1] to calculated the actual printed width.  I just don't know the code well enough to find all the places that need to know the string width/length.

1: https://www.npmjs.com/package/strip-ansi

**Comment by ayonious** [comment]
@abierbaum yea I think thats all that needs to be fixed.

```
 const curTextSize = strip-ansi(text).length; 
```

Please make a PR for faster fix.

**Comment by ayonious** [comment]
fixed in version 2.6

**Comment by abierbaum** [comment]
@ayonious Thank you so much for this!!!  I had it on my list for this weekend but am happy to see you got it first. :)
